### PR TITLE
qa/cephadm: fix bogus mgr deployment

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Previously, we were calling

```
ssh host1 -> cephadm bootstrap ...
ceph orch apply mgr '2;<host2>=x'
```
which instructed cephadm to remove the mgr from host1.

Fixes: https://tracker.ceph.com/issues/48535
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephadm.py [hunk 1]
```diff
@@ -566,22 +592,9 @@ def ceph_mgrs(ctx, config):
...
-                c_, _, id_ = teuthology.split_role(mgr)
-                if c_ == cluster_name and id_ == ctx.ceph[cluster_name].first_mgr:
-                    continue
-                log.info('Adding %s on %s' % (mgr, remote.shortname))
-                nodes.append(remote.shortname + '=' + id_)
-                daemons[mgr] = (remote, id_)
-        if nodes:
-            _shell(ctx, cluster_name, remote, [
-                'ceph', 'orch', 'apply', 'mgr',
-                str(len(nodes) + 1) + ';' + ';'.join(nodes)]
```
Comment (mgfritch) [inline qa/tasks/cephadm.py:583]: any idea why mgr was deployed on +1 whereas the other services are not?

**Review by mgfritch — APPROVED** [review]
generally, lgtm!

File: qa/tasks/cephadm.py (hunk 1 above)
Comment (sebastian-philipp) [reply qa/tasks/cephadm.py:583]: remember the old scheduler algorithm?

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2020-12-11_18:42:23-rados:cephadm-wip-swagner-testing-2020-12-11-1812-distro-basic-smithi/

failed with

```
2020-12-11T19:03:11.472 INFO:teuthology.orchestra.run.smithi176.stderr:--> Zapping: /dev/vg_nvme/lv_4
2020-12-11T19:03:11.472 INFO:teuthology.orchestra.run.smithi176.stderr:Running command: /usr/bin/dd if=/dev/zero of=/dev/vg_nvme/lv_4 bs=1M count=10 conv=fsync
2020-12-11T19:03:11.473 INFO:teuthology.orchestra.run.smithi176.stderr: stderr: 10+0 records in
2020-12-11T19:03:11.473 INFO:teuthology.orchestra.run.smithi176.stderr:10+0 records out
2020-12-11T19:03:11.473 INFO:teuthology.orchestra.run.smithi176.stderr: stderr: 10485760 bytes (10 MB, 10 MiB) copied, 0.0109941 s, 954 MB/s
2020-12-11T19:03:11.473 INFO:teuthology.orchestra.run.smithi176.stderr:--> Zapping successful for: <LV: /dev/vg_nvme/lv_4>
2020-12-11T19:03:12.015 DEBUG:teuthology.orchestra.run.smithi176:> sudo /home/ubuntu/cephtest/cephadm --image quay.ceph.io/ceph-ci/ceph:f8dbeb96eb0e35d1a67a073c06084411ff128972 shell -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring --fsid 21df8f04-3be3-11eb-9811-001a4aab830c -- ceph orch daemon add osd smithi176:vg_nvme/lv_4
```
this command ^^^ simply hangs now for hours.

No idea if this is caused by this one or other PRs.

**Comment by jmolmo** [comment]
The modification seems sensible and i cannot see the relation between the osd creation hung and this .... but still not experience with this tests :-(
Maybe you can add more log messages in order to have more information to debug.

**Comment by sebastian-philipp** [comment]
fixed by #38707

**Pull request closed without merging**